### PR TITLE
290 get redirect url

### PR DESCRIPTION
**Pull request by Thurst-ft**
Work with what kat-client returns not the acquisition service.
Test for module

### CODE REVIEW

File: lib/middleware/assignActiveLicence.js [hunk 1]
```diff
@@ -4,25 +4,28 @@ const logger = require('@financial-times/n-logger').default;
...
+					licence.displayName = acx.displayName || acx.name;
+					licence.signupURI = (acx.barrierContext && acx.barrierContext.redirectUrl)? acx.barrierContext.redirectUrl : null;
 				}
-			}
 
 			return licence;
 		})
 		.catch((err) => {
-			logger.error({operation:'addLicenceAcquisitionData',licenceId: licence.licenceId}, err);
+			logger.error({operation:'addLicenceAcquisitionData',licenceId: licence.licenceId, status:err.name}, err);
```
Comment (leggsimon) [inline lib/middleware/assignActiveLicence.js]: extract `operation` to the top of the function?

File: lib/middleware/assignActiveLicence.js [hunk 2]
```diff
@@ -4,25 +4,28 @@ const logger = require('@financial-times/n-logger').default;
 const RE_FTCOM = /^FT\.com\s*/i;
 
 function _addLicenceAcquisitionData (licence) {
+	licence.displayName = null;
```
Comment (leggsimon) [inline lib/middleware/assignActiveLicence.js]: why is this being null'd?

File: lib/middleware/assignActiveLicence.js [hunk 3]
```diff
@@ -33,14 +36,18 @@ module.exports = (req, res, next) => {
...
 
 	KATConfig.activeLicence = licenceList.find(licence => licence.licenceId === activeLicenceId);
-
+	logger.info({operation:'assignActiveLicence ', licenceObject: JSON.stringify(KATConfig.activeLicence)});
 	if (KATConfig.activeLicence) {
 		return _addLicenceAcquisitionData(KATConfig.activeLicence)
 			.then(result => {
+				logger.info({operation:'assignActiveLicence ', licenceObject: JSON.stringify(result)});
 				KATConfig.activeLicence = result;
+				logger.info({operation:'assignActiveLicence ', licenceObject: JSON.stringify(KATConfig.activeLicence)});
```
Comment (leggsimon) [inline lib/middleware/assignActiveLicence.js]: duplicated logging?

File: lib/middleware/assignActiveLicence.js [hunk 4]
```diff
@@ -33,14 +36,18 @@ module.exports = (req, res, next) => {
...
 				next();
 			})
 			.catch(next);
+	} else {
+		logger.info({operation:'assignActiveLicence ', licenceObject: JSON.stringify(KATConfig)});
```
Comment (leggsimon) [inline lib/middleware/assignActiveLicence.js]: extract `operation` to top of the function?

File: lib/middleware/assignActiveLicence.js [hunk 5]
```diff
@@ -33,14 +36,18 @@ module.exports = (req, res, next) => {
...
+	logger.info({operation:'assignActiveLicence ', licenceObject: JSON.stringify(KATConfig.activeLicence)});
 	if (KATConfig.activeLicence) {
 		return _addLicenceAcquisitionData(KATConfig.activeLicence)
 			.then(result => {
+				logger.info({operation:'assignActiveLicence ', licenceObject: JSON.stringify(result)});
 				KATConfig.activeLicence = result;
+				logger.info({operation:'assignActiveLicence ', licenceObject: JSON.stringify(KATConfig.activeLicence)});
 				next();
 			})
 			.catch(next);
```
Comment (leggsimon) [inline lib/middleware/assignActiveLicence.js]: would it be useful to open this up to log the error?
```js
.catch(err => {
    logger.error({...}, err);
    next(err);
})
```

File: test/lib/middleware/assignActiveLicence.spec.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
+const logger = require('./../../../lib/logger');
+const sinon = require('sinon');
+const expect = require('chai').expect;
+const nock = require('nock');
+const httpMocks = require('node-mocks-http');
+const { assignActiveLicence } = require('./../../../index');
+const { acquisitionCtxClient } = require('@financial-times/kat-client-proxies');
+const licenceContext = require('./../fixtures/licenceContext.js');
+
+describe.only('assignActiveLicence', () => {
```
Comment (leggsimon) [inline test/lib/middleware/assignActiveLicence.spec.js]: remove `.only`

File: test/lib/middleware/assignActiveLicence.spec.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+	});
+
+	afterEach(() => {
+		nock.cleanAll();
+		getContextStub.restore();
+
+	});
+
+	after(() => {
+		nock.cleanAll();
```
Comment (dapenguin) [inline test/lib/middleware/assignActiveLicence.spec.js]: Do we need to `cleanAll` here if we are already doing it in `afterEach`?

File: test/lib/middleware/assignActiveLicence.spec.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+		logMessageStub.restore();
+
+	});
+
+	it('should decorate req.KATConfig.activeLicence with data from acquisition context service ', () => {
+		const res = httpMocks.createResponse();
+		const nextSpy = sinon.spy();
+		const licenceId = '00000000-0000-0000-0000-000000000001';
+		const req = {
```
Comment (dapenguin) [inline test/lib/middleware/assignActiveLicence.spec.js]: Is the `req` object the same in all the tests? If so, declare it once.

File: lib/middleware/assignActiveLicence.js (hunk 2 above)
Comment (Thurst-ft) [reply lib/middleware/assignActiveLicence.js]: Wanted to make sure the object has the same keys regardless of if anything is returned from the acquisition context. Also wanted to be explicit about a value in that scenario. I've changed to check for it's existence first so null is only set if an empty key is added.

File: test/lib/middleware/assignActiveLicence.spec.js (hunk 7 above)
Comment (Thurst-ft) [reply test/lib/middleware/assignActiveLicence.spec.js]: nope

File: test/lib/middleware/assignActiveLicence.spec.js (hunk 6 above)
Comment (Thurst-ft) [reply test/lib/middleware/assignActiveLicence.spec.js]: 🤦‍♂️

**Comment by Thurst-ft** [comment]
@dapenguin @leggsimon suggested changes made please can you review. Thanks

File: lib/middleware/assignActiveLicence.js [hunk 9]
```diff
@@ -4,24 +4,23 @@ const logger = require('@financial-times/n-logger').default;
 const RE_FTCOM = /^FT\.com\s*/i;
 
 function _addLicenceAcquisitionData (licence) {
-	licence.displayName = null;
+	const operation = 'addLicenceAcquisitionData';
+	licence.displayName = licence.displayName ? licence.displayName: null;
 	licence.productAbbrv = licence.product ? licence.product.replace(RE_FTCOM, '') : '';// no need for `ft.com` in the abbr;
-	licence.signupURI = null;
+	licence.signupURI = licence.signupURI? licence.signupURI : null;
```
Comment (dapenguin) [inline lib/middleware/assignActiveLicence.js]: Picky, but space before `?`

**Pull request merged**